src/env.rs

```diff
@@ -101,7 +100,11 @@ pub fn get_partition_ctx(
     yb8: c_int,
     xb8: c_int,
 ) -> u8 {
-    (a.partition[xb8 as usize] >> (4 - bl) & 1) + ((l.partition[yb8 as usize] >> (4 - bl) & 1) << 1)
+    // the right-most ("index zero") bit of the partition represents the 8x8 block level,
+    // but the BlockLevel enum represents the variants numerically in the opposite order
+    // (128x128 = 0, 8x8 = 4). The shift corrects the ordering.
+    let shift = 4 - bl as u8;
+    ((a.partition[xb8 as usize] >> shift) & 1) + (((l.partition[yb8 as usize] >> shift) & 1) << 1)
```

Comment:
would this be better?
```suggestion
    let get = |partition: &[_], index| partition[index as usize] >> (4 - bl as u8) & 1;
    get(&a.partition.0, xb8) + get(&l.partition.0, yb8) << 1
```

Reply:
I think either way is fine. The small bit of duplicated logic between the two isn't really an issue imo, but I think the closure approach is also nice because it makes it clear that the logic is the same for both of them.

Reply:
It may be simpler to define `let has_bl = |x| (x >> (4 - bl as u8)) & 1;` and then use as `has_bl(a.partition[xb8 as usize]) + has_bl(b.partition[yb8 as usize]) * 2`. Note that I generally prefer using `* 2` instead of `<< 1` in such cases to avoid adding extra parentheses. Also the precedence of `*` is generally much better understood than the precedence of `<<`.

Reply:
yes, good call!